### PR TITLE
Fix bug where speechbrain would request files from HF even with local files provided

### PR DESCRIPTION
**Pull request by gabrielziegler3**
#1253  Signed-off-by: Gabriel Ziegler <gabrielziegler3@gmail.com>

### CODE REVIEW

**Comment by mravanelli** [comment]
@Gastron

**Comment by gabrielziegler3** [comment]
I believe that this change alone doesn't fix the whole issue, but the lines I changed were the one making `fetch` fallback to requesting from `Hugging Face`.

**Comment by mravanelli** [comment]
I'm fine with these changes. 
- As for the gradient accumulation, we already considered adding it in the `fit_batch` function of core.py.  However, as you can see there is often the need to override the `fit_batch` method within the recipe script. For instance, in the [CommonVoice Transformer recipe](https://github.com/speechbrain/speechbrain/blob/develop/recipes/CommonVoice/ASR/transformer/train.py#L150), we need to override it because we have a learning rate annealing done at every update (not supported by the version in `core.py`). 
In this case, we have to re-implement the gradient accumulation again in train.py.

- As it is basically very hard to write a` fit_batch` function that meets everyone's needs, our approach was to keep in `core.py` a basic version of it and leave users all the freedom to customize  ` fit_batch`  based on their needs in train.py. Gradient descends only require a couple of lines of code, so it shouldn't be too hard to add.  Said that I'm fine adding it `core.py` as well if this is considered useful by the users.

So @Gastron, there is not need to update the existing recipes because they already have to override fit_batch.

**Comment by mravanelli** [comment]
What do you think of this PR @Gastron?

**Comment by Adel-Moumen** [comment]
Up @Gastron, could you please have a look ?

**Comment by Gastron** [comment]
Ooh this is an old PR, has somehow slipped past me, thanks for bringing it up again.

**Review by Gastron — COMMENTED** [review]
This changes the if/else logic, and would unfortunately break the case where `default_source` is not specified.

It seems to me that this PR just stems from path specification that did not work. I wonder if it is a path like `"./filename.ckpt"`, perhaps that is currently broken?
